custom_components/tahoma/sensor.py

```diff
@@ -111,47 +124,43 @@ def update(self):
 
         self.controller.get_states([self.tahoma_device])
 
-        if CORE_LUMINANCE_STATE in self.tahoma_device.active_states:
-            self.current_value = self.tahoma_device.active_states.get(
-                CORE_LUMINANCE_STATE
-            )
+        states = self.tahoma_device.active_states
 
-        if CORE_RELATIVE_HUMIDITY_STATE in self.tahoma_device.active_states:
+        if CORE_LUMINANCE_STATE in states:
+            self.current_value = states.get(CORE_LUMINANCE_STATE)
+
+        if CORE_RELATIVE_HUMIDITY_STATE in states:
             self.current_value = float(
-                "{:.2f}".format(
-                    self.tahoma_device.active_states.get(CORE_RELATIVE_HUMIDITY_STATE)
-                )
+                "{:.2f}".format(states.get(CORE_RELATIVE_HUMIDITY_STATE))
```

Comment:
We should use f-strings here. I think this will be required for a merge in core.
`f'{states.get(CORE_RELATIVE_HUMIDITY_STATE):%.2f}'`

Reply:
@tetienne do we need fstrings for all floats here? https://github.com/home-assistant/core/pull/36258#issuecomment-637720333

Reply:
What do you have in entry and what do you want in output?

To have a float from a string you can do:
```
>>>a = "2.3"
>>>float(a)
2.3
```

Reply:
do we actually convert a float to string then to float?

Reply:
To be honest, I am not sure how HA would like to have the values. Do they require a float or is a floating value in a string the right approach?

In the end, what should happen in `sensor.py` is that we make round the value and make it an int or float.

Reply:
I think a float is good enough. but why not do `round(states.get(CORE_RELATIVE_HUMIDITY_STATE),2)` instead of converting to a string for the rounding?

Reply:
and I would do the same in sensor.py

Reply:
@vlebourl, is states.get(CORE_RELATIVE_HUMIDITY_STATE) already a float? If a string, the round method can not be applied.
```
>>> a = "2.3"
>>> round(a, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: type str doesn't define __round__ method

Reply:
I think it. otherwise we would get
```
>>> "{:.2f}".format("2.45")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
ValueError: Unknown format code 'f' for object of type 'str'
```

Reply:
And that's the signature for my temperature sensor:
![image](https://user-images.githubusercontent.com/5701372/86593232-1fa28b80-bf95-11ea-9143-9340ff770c9f.png)

Reply:
Let's maybe leave this like this for now as this PR is already quite large, but resolve this in another PR once this one is merged

Reply:
Agree, let's work on those improvements in separate smaller PR's. This one already got way to big :D